labs/direct-state-transfer_perun.md

```diff
@@ -0,0 +1,62 @@
+# Lab Name
+
+Direct State Transfer / Perun
+
+# Short Description
+
+Blockchain-agnostic State Channels framework, written in Go.
+
+# Scope of Lab
+
+Perun / Direct State Transfer is a Go implementation of the [Perun state
+channel protocols](https://perun.network/) ([introduction
+paper](https://perun.network/pdf/Perun2.0.pdf)). The Perun protocols provide
+payment and general state channel functionality to all existing blockchains
+that feature smart contracts. As a blockchain scalability solution, payment
+and state channels reduce transaction costs and increase the system throughput
+by executing incremental transactions off-chain. The Perun protocols have been
+proven cryptographically secure in the UC-framework. They are
+blockchain-agnostic and only rely on a blockchain's capability to execute
+smart contracts.
+
+Currently the project consists of three main parts:
+
+- _go-perun_ is a Go SDK that implements the core state channel protocol in a
+  blockchain-agnostic fashion by following the dependency inversion principle.
+  For this reason, several blockchain backends can be chosen and only
+  blockchain-specific initializations need to be executed at program startup.
+
+- _contracts-eth_ contains the interconnected smart contracts for the
+  Ethereum backend. They define the on-chain logic for channel deposits,
+  disputes, settlements and withdrawals.
+
+- _dst-go_ is a state channel node. It builds upon the go-perun SDK for
+  establishing and operating Perun state channels, sets up a network
+  for off-chain communications and provides an API interface for the user to
+  interact with. To start with, Ethereum payment channels are implemented.
+
+# Initial Committers
+
+- <https://github.com/anagha-ks>
+- <https://github.com/choeppler>
+- <https://github.com/danielksan81>
+- <https://github.com/ggwpez>
+- <https://github.com/Gold3nboy>
+- <https://github.com/manoranjith>
+- <https://github.com/matthiasgeihs>
+- <https://github.com/ndzik>
+- <https://github.com/RmbRT>
+- <https://github.com/sebastianst>
+- <https://github.com/tinnendo>
+
+# Sponsor
+
+- <https://github.com/lehors> - Chair of the TSC
+
+# Pre-existing repository
```

Comment:
At least the go-perun repository does not have DCO signoffs on all commits. DCO signoffs will be required if we were to transfer the repos to Hyperledger Labs. Can you comment on your expectation for transferring these repos, as well as, whether you would want five separate repos? Or would you only like a single repository that you will create sub-directories under and manually bring the code from these repos into?

Reply:
The [contracts-eth](https://github.com/direct-state-transfer/contracts-eth) repository is currently in the process of being re-licensed to Apache-2.0.

As for the Signed-off-by-requirement: the three [dst-* repositories](https://github.com/direct-state-transfer?q=dst-d&type=&language=) should be fine, and we planned to add the signoffs to [go-perun](https://github.com/direct-state-transfer/go-perun) and [contracts-eth](https://github.com/direct-state-transfer/contracts-eth) as soon as the proposal gets accepted so all of the repositories can be transferred. Is that okay?

And yes, we'd like to keep the separate repositories.

Reply:
Licensing is important. All code will need to be Apache 2.0.

As long as signoffs exist before the transfer, that would be acceptable.

Reply:
@tkuhrt Would it be ok to leave the SDK at its current name `go-perun` and prefix all other repos just with `perun-`? It is custom for Go repos/projects to name them `go-...`, like `go-ethereum` etc. so we'd prefer to leave the SDK as `go-perun`. This way we also wouldn't need to rename it in several places where it is already used. The other repos can start with `perun-...`. Also, `perun-framework-` would be a very long and cumbersome prefix so `perun-` would be preferable. E.g. `perun-contracts-eth` is already long but `perun-framework-contracts-eth` is not nice IMO.